stdlib/public/core/FloatingPointTypes.swift.gyb

```diff
@@ -83,15 +83,29 @@ extension ${Self} : CustomStringConvertible {
   /// A textual representation of the value.
   @_inlineable // FIXME(sil-serialize-all)
   public var description: String {
-    return _float${bits}ToString(self, debug: false)
+    if isFinite {
+      return _float${bits}ToString(self, debug: false)
+    } else if isNaN {
+      return "nan"
+    } else if sign == .minus {
+      return "-inf"
+    } else {
+      return "inf"
```

Comment:
If we short-circuit inf here, we probably don't need to worry about it at the next level down.

Reply:
Removed infinity handling in the next layer down.